### PR TITLE
Add collegiate subreddits list page

### LINKED ISSUE #7: Implement List page for CollegiateSubreddits
Implement List page for CollegiateSubreddits 

Implement the List page using the Table component made in a previous issue.  For now, no need for the Edit and Delete columns (those will come later). 

You may use the UCSBStudentsIndexPage as a model.

- [x] When loading the page, it fetches data from the backend and populates the table
- [x] test coverage is 100% for `npm run coverage`
- [x] test coverage is 100% for `npx stryker run`

### PR DESCRIPTION
**Pull request by SwiftWinds**
# Overview

This adds the UI for the collegiate subreddits list page (without edit/delete options)

# Issues Addressed

Closes #7 

# Details

Where applicable, include "before" and "after" screenshots or animated GIFs.   The tool [LiceCAP](https://www.cockos.com/licecap/) or one of [these similar tools](https://www.nextofwindows.com/5-free-tools-to-screen-capture-to-gif-on-windows) 
can be used to capture screenshots/animated GIFs.

## Reminders (delete this section before merging or as code is merged)
- [ ] Explicitly link to this pull request each issue that it closes
- [ ] Assign one and ideally two reviewers to review this pull request. 
- [ ] Reviewers should not just approve with "Looks good!" Instead, reviewers should carefully go through the code and use GitHub's reviewer interface to ask questions, make comments, and suggest changes.

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -14,84 +14,96 @@ import UCSBDatesEditPage from "main/pages/UCSBDates/UCSBDatesEditPage";
 import StudentsIndexPage from "main/pages/Students/StudentsIndexPage";
 import StudentsCreatePage from "main/pages/Students/StudentsCreatePage";
 
-import UCSBSubjectsIndexPage from "main/pages/UCSBSubjects/UCSBSubjectsIndexPage";
-import UCSBSubjectsCreatePage from "main/pages/UCSBSubjects/UCSBSubjectsCreatePage";
+import CollegiateSubredditsIndexPage from "main/pages/CollegiateSubreddits/CollegiateSubredditsIndexPage";
```
Comment (nguglielmin) [inline frontend/src/App.js:20]: did you delete the import statements for `UCSBSubjects`? If this is in `App.js` they need to stay right?

File: frontend/src/App.js [hunk 2]
```diff
@@ -14,84 +14,96 @@ import UCSBDatesEditPage from "main/pages/UCSBDates/UCSBDatesEditPage";
...
+            <Route
+              exact
+              path="/collegiateSubreddits/create"
+              element={<CollegiateSubredditsCreatePage />}
+            />
+          </>
+        )}
       </Routes>
     </BrowserRouter>
   );
```
Comment (nguglielmin) [inline frontend/src/App.js:135]: revert `App.js` commits and then add your changes back in (ie don't delete anything)

**Review by nguglielmin — CHANGES_REQUESTED** [review]
need to revert changes to `App.js` to not write over tyler's commit.

File: frontend/src/App.js (hunk 2 above)
Comment (SwiftWinds) [reply frontend/src/App.js:135]: Done! Good catch

File: frontend/src/App.js (hunk 1 above)
Comment (SwiftWinds) [reply frontend/src/App.js:20]: Finished this 👍 Thanks!

**Review by nguglielmin — APPROVED** [review]
requested changes have been implemented -- lgtm!

**Pull request merged**